Mathlib/Data/Fintype/Pi.lean

```diff
@@ -163,3 +163,21 @@ theorem Finset.univ_pi_univ {α : Type*} {β : α → Type*} [DecidableEq α] [F
     (Finset.univ.pi fun a : α => (Finset.univ : Finset (β a))) = Finset.univ := by
   ext; simp
 #align finset.univ_pi_univ Finset.univ_pi_univ
+
+/-! ### Diagonal -/
+
+namespace Finset
+variable {ι : Type*} [DecidableEq (ι → α)] {s : Finset α} {f : ι → α}
+
+/-- The diagonal of a finset `s : Finset α` as a finset of functions `ι → α`. -/
+def piDiag (s : Finset α) (ι : Type*) [DecidableEq (ι → α)] : Finset (ι → α) := s.image (const ι)
```

Comment:
Does `.map` work here?

Reply:
Subtly, no: If `\iota` is empty, then `const \iota` is not injective. Of course in that case the finset is obvious to construct, but I can't case on whether `\iota` is empty or not (unless I assume `Decidable (IsEmpty \iota)` which is.. unorthodox)

Reply:
Does the empty case make any sense in your application?

Reply:
It would be kind of strange as my application is `\iota := Fin n` and I would have to rule out `n = 0` explicitly. Not the end of the world, though.